filters/apiusagemonitoring/filter.go

```diff
@@ -70,49 +154,93 @@ func (f *apiUsageMonitoringFilter) String() string {
 	return fmt.Sprintf("%T %s", f, js)
 }
 
-// getMetricNames returns the structure with names of the metrics for this specific context.
-// If it is not already cached, it is generated and cached to speed up next calls.
-func (f *apiUsageMonitoringFilter) getMetricNames(req *http.Request) *metricNames {
-
-	// Match the path to a known template
-	var path *pathInfo = nil
+// resolvePath tries to match the request's path with one of the configured path template.
+func (f *apiUsageMonitoringFilter) resolvePath(req *http.Request) *pathInfo {
 	for _, p := range f.Paths {
 		if p.Matcher.MatchString(req.URL.Path) {
-			path = p
-			break
+			return p
 		}
 	}
-	if path == nil {
-		path = unknownPath
-	}
+	return f.Spec.unknownPath
+}
 
-	// Get the cached prefixes for this path and verb
+// getEndpointMetricsNames returns the structure with names of the metrics for this specific context.
+// It tries first from the path's cache. If it is not already cached, it is generated and
+// caches it to speed up next calls.
+func getEndpointMetricsNames(req *http.Request, path *pathInfo) *endpointMetricNames {
 	method := req.Method
 	methodIndex, ok := methodToIndex[method]
 	if !ok {
-		methodIndex = MethodIndexUnknown
+		methodIndex = methodIndexUnknown
 		method = unknownElementPlaceholder
 	}
 
 	prefixes := path.metricPrefixesPerMethod[methodIndex]
```

Comment:
I would rather do it similar as above:

```go
prefixes, ok := path.metricPrefixesPerMethod[methodIndex]
if ok { return prefixes }
return createAndCacheMetricsNames(path, method, methodIndex)
```

Reply:
This is an `array`, not a `map`. There is no `ok` 2nd return value.

Reply:
I could do something like 

```go
	if p := path.metricPrefixesPerMethod[methodIndex]; p != nil {
		return p
	}
	return createAndCacheMetricsNames(path, method, methodIndex)
```

to be more go-ish.